v0/destinations/marketo/util.js

```diff
@@ -126,50 +125,46 @@ const sendPostRequest = async (url, data, options) => {
   return processedResponse;
 };
 
-// eslint-disable-next-line no-unused-vars
-const responseHandler = (destinationResponse, _dest) => {
-  const message = `[Marketo Response Handler] - Request Processed Successfully`;
-  const { status } = destinationResponse;
-  // if the responsee from destination is not a success case build an explicit error
-  if (!isHttpStatusSuccess(status)) {
-    throw new ErrorBuilder()
-      .setStatus(status)
-      .setMessage(
-        `[Marketo Response Handler] - Request failed  with status: ${status}`
-      )
-      .setDestinationResponse(destinationResponse)
-      .isTransformResponseFailure(true)
-      .setStatTags({
-        destination: DESTINATION,
-        stage: TRANSFORMER_METRIC.TRANSFORMER_STAGE.RESPONSE_TRANSFORM,
-        scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.SCOPE,
-        meta: getDynamicMeta(status)
-      })
-      .build();
+class MarketoNetworkHandler extends GenericNetworkHandler {
+  // eslint-disable-next-line class-methods-use-this
+  responseHandler(destinationResponse) {
+    const message = `[Marketo Response Handler] - Request Processed Successfully`;
+    const { status } = destinationResponse;
+    // if the responsee from destination is not a success case build an explicit error
+    if (!isHttpStatusSuccess(status)) {
+      throw new ErrorBuilder()
+        .setStatus(status)
+        .setMessage(
+          `[Marketo Response Handler] - Request failed  with status: ${status}`
+        )
+        .setDestinationResponse(destinationResponse)
+        .isTransformResponseFailure(true)
+        .setStatTags({
+          destination: DESTINATION,
+          stage: TRANSFORMER_METRIC.TRANSFORMER_STAGE.RESPONSE_TRANSFORM,
+          scope: TRANSFORMER_METRIC.MEASUREMENT_TYPE.API.SCOPE,
+          meta: getDynamicMeta(status)
+        })
+        .build();
+    }
+    // check for marketo application level failures
+    marketoApplicationErrorHandler(
```

Comment:
Why are we naming methods against the destination? Also, we are exporting it separately. Shouldn't the exported class take care of that?

Reply:
yes we should maybe refactor it further. The problem of keeping those specific methods in class is that, then an instance needs to be created in `transform.js` to use that. We can think of `marketoApplicationErrorHandler` as more of a utility function being used in transform.js.  
I am not sure if we should be looking towards creating instances of this class elsewhere other than the factory (at least as of now).
@utsabc  wdyt ? I think @saikumarrs  has a good point.

Reply:
IMO, it should be the factory's responsibility to instantiate and cache these objects. On a tangent, should these be converted to static classes (not even singleton), if they don't really hold any "instance" specific data